devito/passes/clusters/aliases.py

```diff
@@ -1130,3 +1166,35 @@ def wset(exprs):
     """
     return {i.function for i in flatten([e.free_symbols for e in as_tuple(exprs)])
             if i.function.is_AbstractFunction}
+
+
+def potential_max_deriv_order(exprs):
+    """
+    The maximum FD derivative order in a list of expressions.
+    """
+    # NOTE: e might propagate the Derivative(...) information down from the
+    # symbolic language, but users may do crazy things and write their own custom
+    # expansions "by hand" (i.e., not resorting to Derivative(...)), hence instead
+    # of looking for Derivative(...) we use the following heuristic:
+    #   add(mul, mul, ...) -> stems from first order derivative
+    #   add(mul(add(mul, mul, ...), ...), ...) -> stems from second order derivative
```

Comment:
Not completely convinced by this heuristic eventhough I think it makes sense but may miss some cases (i.e pure `Derivative(Derivative(expr))`)